src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -6428,15 +6428,41 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
         }
 
         if (!s) {
-            std::set<std::string> not_cpython_builtin = {
-                "sin", "cos", "gamma", "tan", "asin", "acos", "atan"
+            std::set<std::string> module_math = {
+                "sin", "cos", "gamma", "tan", "asin", "acos", "atan", "exp"
             };
-            if (ASRUtils::IntrinsicFunctionRegistry::is_intrinsic_function(call_name)
-             && not_cpython_builtin.find(call_name) == not_cpython_builtin.end()) {
+            std::set<std::string> module_numpy = {
+                "exp", "exp2"
+            };
+            // Complex conditions because of functions which are in both modules
+            bool is_math_imported = (current_scope->resolve_symbol("math") != nullptr) ? true : false;
```

Comment:
The condition is also boolean so you don't need to use a conditional operator which again produces boolean.

Reply:
Though the condition is fragile. What if I as a user also create a module named `math`? It would be true in that case as well.